R/guide-colorbar.r

```diff
@@ -119,11 +128,18 @@ guide_colourbar <- function(
   default.unit = "line",
   reverse = FALSE,
   order = 0,
+  border = element_line(color = 'white', size=1, linetype = 'solid', lineend = 'butt'),
 
   ...) {
 
   if (!is.null(barwidth) && !is.unit(barwidth)) barwidth <- unit(barwidth, default.unit)
   if (!is.null(barheight) && !is.unit(barheight)) barheight <- unit(barheight, default.unit)
+  
+  # if logical argument supplied to tick then use defaults
+  if (identical(ticks, TRUE))
```

Comment:
You're currently specifying the default in two places - here and above in the parameter list. I think it would be better to specify only in one place.

I'd recommend restructuring this if block to cover all cases:

```R
if (identical(ticks, TRUE)) { 

} else if (identical(ticks, FALSE)) {

} else if (inherits(ticks, "element")) {

} else {
  stop("`ticks` should be ...", call. = FALSE)
}
```